### DISCUSSION TITLE
FreeBSD with kernel-libipsec cannot install route(virtual IP) on server side

### DISCUSSION THREAD

**Discussion by lpc-v** [strongswan/strongswan#1499] [Q&A]
I have two FreeBSD machines.
192.168.1.0 -- 10.0.0.1 === 10.0.0.2 -- 192.168.2.0
                          moon             sun

moon and sun both enable `kernel-libipsec` for unsupported algorithm

On server there is an ip pool, and I set vips = 0.0.0.0 on client.

It appears that client `sun` works fine. Client create `tun0` and `tun1`, `tun1` installed the `virtual ip` from ip pool which I set on `moon`.
But the `moon` just has one TUN device `tun0` and `tun0` has no ipv4 address. So intall route on dev `tun0` faild. It reports `adding PF_ROUTE route failed: Network is unreachable`.

So I am wondering is it OK for FreeBSD with `kernel-libipsec` serve as a server gateway? Or vitual IP can be installed on server side? Or I have missed some configuration?

**Comment by tobiasbrunner** [comment]
> moon and sun both enable `kernel-libipsec` for unsupported algorithm

What does that mean?



> It reports `adding PF_ROUTE route failed: Network is unreachable`.

If you [increase the log level](https://docs.strongswan.org/docs/5.9/config/logging.html) for `knl` to 2 you see what route the daemon tried to install. Not sure if the problem is that the destination is a `/32` virtual IP address (or from which subnet it was allocated). Might depend on the other routes on your system, so check that.

> So I am wondering is it OK for FreeBSD with `kernel-libipsec` serve as a server gateway?

It's definitely nothing we recommend (see [here](https://docs.strongswan.org/docs/5.9/plugins/kernel-libipsec.html) for details).

> Or vitual IP can be installed on server side?

Not on the TUN device. You could manually install a virtual IP but I don't think that would make a difference.

> Or I have missed some configuration?

Maybe, but since you haven't posted any of the information requested in #196 it's impossible to tell.
